webapp/shop/cred/views.py

```diff
@@ -293,76 +301,78 @@ def cred_exam(trueability_api, **_):
 
 @shop_decorator(area="cred", permission="user", response="html")
 @canonical_staff()
-def cred_provision(trueability_api, **_):
+def cred_provision(ua_contracts_api, trueability_api, **_):
+    contract_item_id = flask.request.args.get("contractItemID", type=int)
+
+    if contract_item_id is None:
+        return flask.redirect("/credentials/your-exams")
+
     sso_user = user_info(flask.session)
-    sso_user_email = sso_user["email"]
-    ability_screen_id = 4194
+    country_code = get_user_country_by_ip().json["country_code"] or "DE"
```

Comment:
Do we want to default to "DE"?

Reply:
@jpmartinspt get_user_country_by_ip() may set `country_code` to `None` if the lookup failed. The country code is a mandatory field to create an exam reservation (and is used to determine the deployment region). Would it be better show an error, or use a default?

Reply:
Hmm I think it's fine to default to a specific region, the question was more if "DE" was indeed the optimal default.

Reply:
If there is no drawback in doing so maybe we could let the user pick the region.(We don't need to block this PR for that now)

Reply:
Cool, we can do another PR to let the user select their country if we can't look it up.

For this PR, maybe "GB" is a more suitable default since Canonical is UK-based? (Also, one of our Pools is configured to use europe-west2 (London).)